### ISSUE TITLE
Some games do not load properly on UV V3

### ISSUE THREAD

**Issue by qerionx** [titaniumnetwork-dev/Ultraviolet#158]
Issue tracker is **ONLY** used for reporting bugs. New features should be discussed on our Discord server.

## Expected Behavior
All games are able to load successfully and you are able to play as expected.

## Current Behavior
**Florr.io, Digdig.io, beta.deeeep.io** Stuck on connecting screen, cannot connect

**Territorial.io:** Works on singleplayer, but stuck on connecting screen on multiplayer

~~**Subway Surfers:** Downloads required assets, but fails to load.~~ _Uses Unity_

**All games using Unity 2017+:** Fails to load.

**Eaglercraft 1.x:** Multiplayer does not connect

## Possible Solution
It works with older versions of Ultraviolet (not V3)\
I don't know what causes this issue, but it seems to me that for the multiplayer games not connecting, it could be the same issue that causes that to happen, and for the Unity games not loading, that could be a separate issue

## Steps to Reproduce

~~1. Go to any online [subway surfers](https://files.gamezhero.com/game/905/1d1/9051d1173be765fb/data/index.html) link (it downloads the same assets from the same source) it will download but it will be stuck.~~
2. Go to https://florr.io, and select "play as guest." It will stay on "Connecting..."
3. Go to https://digdig.io and select "play." it will stay on "Connecting..."
4. Go to https://beta.deeeep.io/, you are unable to play due to it trying to connect
5. Go to https://territorial.io and select Multiplayer. It will stay on a loading screen.
6. Go to a unity game. (see this [website](https://unblocked-games.s3.amazonaws.com/unity/index.html)) click on a game, and it will not load.
7. Go to Eaglercraft 1.x (see [website](https://derpmandev.github.io/unblocked-games/eaglercraft-1-5/)) go to multiplayer, and servers will not be able to connect. 

## Context (Environment)
Many users have been reporting this issue which is why I'm making this issue on the GitHub
Again, this worked before I upgraded to UV V3. I would tell you what causes this issue but I honestly don't know, I'm not sure if it's the same issue that causes these websites to not function properly, or if there are multiple issues. The websites that are listed are not the only ones that don't work. I have noticed that a lot of online multiplayer games fail to connect as well.

If this could be fixed, it would be greatly appreciated.

## Additional info
May be unrelated, but the text & icons on Eaglercraft seem very glitchy, **similar to Spotify on UV V3**

<img src="https://github.com/user-attachments/assets/5f9aaddd-5d50-4738-98e2-d88dfdecca2b" width="300">
<img src="https://github.com/user-attachments/assets/1027a5d5-6a1a-48b8-abe3-02d46b5b1589" width="300">

.

Additionally, Ultraviolet might have problems with connecting to websockets which might cause some multiplayer games to not work. See this image below. (Websocket tests return working which is weird)

<img src="https://github.com/user-attachments/assets/2f35f523-8af0-4046-bf13-c4b2a1c30a70" width="300">

**Comment by DerpmanDev** [comment]
Also, I'm not sure if the transport has anything to do with this, but I'm using epoxy

**Comment by quotingg** [comment]
not really sure what causes the subway surfers link but the other two are due to your bare server websocket

**Comment by DerpmanDev** [comment]
> not really sure what causes the subway surfers link but the other two are due to your bare server websocket

What do you mean? How should I fix it?

**Comment by quotingg** [comment]
> > not really sure what causes the subway surfers link but the other two are due to your bare server websocket
> 
> What do you mean? How should I fix it?

you should host your bare server on a different platform but before that what are you using to host your bare server?

**Comment by DerpmanDev** [comment]
I don't use a bare server, it's Wisp

**Comment by DerpmanDev** [comment]
> > Also, I'm not sure if the transport has anything to do with this, but I'm using epoxy
> > Update: Most multiplayer games don't work on Ultraviolet V3 (it works on older versions) A lot of websites that use websockets try to connect to wss://`<proxy-site>`/`<path>` instead of using the game link, which breaks it
> 
> i cant reproduce with territorial or florr as both work fine on my end, the subway surfers thing is weird but ill look into it

Do you think it can be something with my server? I've tested it on multiple proxies running V3 and all of them seem to have the same error

**Comment by quotingg** [comment]
> I don't use a bare server, it's Wisp

could probably be something with the version of wisp but other than that i have no clue

**Comment by DerpmanDev** [comment]
It also seems that any game that is running Unity 2017+ does not seem to load.

See this website to reproduce the error: https://unblocked-games.s3.amazonaws.com/unity/index.html

**Comment by Percslol** [comment]
check the actual uv version you are using

**Comment by Percslol** [comment]
eaglercraft 1.x works fine, the relays connecting here is proof enough that it works (if you cant connect to the relays, you would probably not be able to make a server connection either), you just have a weird build of it or uv
![image](https://github.com/user-attachments/assets/619f5697-9fd6-4e1d-97a5-bec7ff7c5630)

territorial.io works fine, doesn't get stuck on connecting.
![image](https://github.com/user-attachments/assets/8c0de1d7-8ef2-4684-a297-a1ee563b6b12)

**Comment by DerpmanDev** [comment]
> eaglercraft 1.x works fine, the relays connecting here is proof enough that it works (if you cant connect to the relays, you would not be able to make a server connection either), you just have a weird build of it or uv ![image](https://private-user-images.githubusercontent.com/83934299/360598186-619f5697-9fd6-4e1d-97a5-bec7ff7c5630.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ0MzkxMTEsIm5iZiI6MTcyNDQzODgxMSwicGF0aCI6Ii84MzkzNDI5OS8zNjA1OTgxODYtNjE5ZjU2OTctOWZkNi00ZTFkLTk3YTUtYmVjN2ZmN2M1NjMwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIzVDE4NDY1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI5NWNmMTMyZTVhOWFhNDQzNTJmOWU2OWQwNzVlNDA5MjcwZDhmNjU5M2MzMWQxZDM1ZjVkY2ZjMDc5YzcxYmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.l3IhMnFh_dy6HbO6anGMINy6NsSkzBfUZYnsKrJTv84)
> 
> territorial.io works fine, doesn't get stuck on connecting. ![image](https://private-user-images.githubusercontent.com/83934299/360598611-8c0de1d7-8ef2-4684-a297-a1ee563b6b12.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ0MzkxMTEsIm5iZiI6MTcyNDQzODgxMSwicGF0aCI6Ii84MzkzNDI5OS8zNjA1OTg2MTEtOGMwZGUxZDctOGVmMi00Njg0LWEyOTctYTFlZTU2M2I2YjEyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIzVDE4NDY1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZkNzM5MDA3OTQzOTE0M2M5ODJlOWZlNjBjMmMzZTJhZThkZWMxOWRkYzIwMDcxZWY0Njc5MGUxZGZiZjAzZDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.s2NRMWMinFbbK0QMDvrHu8Ejk08RJcTpMlEhnmwUO9U)

Hey Percs, are you running UV locally, or on a server? The errors I am getting are happening on a server. It may be different if it is locally hosted, I don't know.

I'm using UV v3.2.6, and it's imported from the npm package. Not sure if the transport matters, but I'm using epoxy v2.1.5 with bare-mux v2.0.4.

**Comment by Percslol** [comment]
@DerpmanDev issue has been traced, unity 2017 games work, eaglercraft and territorial werent broken, and im still trying to figure out the issue with floor.io and digdig, its probably something completely unrelated since deeeep works fine

**Comment by Percslol** [comment]
the icon issue was caused due to a css rewriter bug, which im actively fixing atm

**Comment by DerpmanDev** [comment]
> @DerpmanDev issue has been traced, unity 2017 games work, eaglercraft and territorial werent broken, and im still trying to figure out the issue with floor.io and digdig, its probably something completely unrelated since deeeep works fine

Got it thank you
